### ISSUE TITLE
A flaky failure in hive-tests setup 

### ISSUE THREAD

**Issue by elharo** [prestodb/presto#22140]
This one isn't a failure in a specific test, but rather a failure when trying to launch hive tests / hive-tests (pull_request)

2024-03-08T20:30:51.5495195Z Creating conf_hadoop-master_1 ... error
2024-03-08T20:30:51.5498499Z 
2024-03-08T20:30:51.5502068Z ERROR: for conf_hadoop-master_1  Cannot start service hadoop-master: driver failed programming external connectivity on endpoint conf_hadoop-master_1 (b89f75e51118e98a7f8b8021eafabdfa0b1018fe1e83f550d3813e3b4b976d62): Error starting userland proxy: listen tcp4 0.0.0.0:50070: bind: address already in use
2024-03-08T20:30:51.5505441Z 
2024-03-08T20:30:51.5507957Z ERROR: for hadoop-master  Cannot start service hadoop-master: driver failed programming external connectivity on endpoint conf_hadoop-master_1 (b89f75e51118e98a7f8b8021eafabdfa0b1018fe1e83f550d3813e3b4b976d62): Error starting userland proxy: listen tcp4 0.0.0.0:50070: bind: address already in use
2024-03-08T20:30:51.5511718Z Encountered errors while bringing up the project.
2024-03-08T20:30:51.6001352Z ##[error]Process completed with exit code 1.

**Comment by arhimondr** [comment]
Looking

**Comment by aaneja** [comment]
The error -
```
 Error starting userland proxy: listen tcp4 0.0.0.0:50070: bind: address already in use
```
suggests that docker compose failed to shutdown and start the hadoop container cleanly, as [part of the test setup](https://github.com/prestodb/presto/blob/69ec7605a1e4e373dcfc6b40b0149a673cedb8b4/presto-hive-hadoop2/bin/run_hive_tests.sh#L7-L8)

I will experiment with some backoff-retries to make this less flaky

**Comment by tdcmeehan** [comment]
Fixed by #22293
